src/Commands/core/ArchiveDumpCommands.php

```diff
@@ -166,9 +168,48 @@ private function createArchiveFile(array $archiveComponents, array $options): st
         $archivePath = Path::join(dirname($this->archiveDir), self::ARCHIVE_FILE_NAME);
 
         stream_wrapper_restore('phar');
+        $this->logger()->info(var_export($archivePath, true));
         $archive = new PharData($archivePath);
 
         $this->createManifestFile($options);
+
+        $this->logger()->info(var_export($this->archiveDir, true));
+
+        // If symlinks are disabled, convert symlinks to full content.
+        if (is_dir($this->archiveDir)) {
+            $this->logger()->info(dt('Converting symlinks...'));
+
+            $iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($this->archiveDir), RecursiveIteratorIterator::SELF_FIRST);
+
+            foreach ($iterator as $file) {
```

Comment:
It would be helpful to break this into a separate function, so that we could have a docblock explaining the purpose. The 'convert-symlinks' could be passed in as a named parameter (rather than passing all of $options).